backend/imagekit/imagekit.go

```diff
@@ -0,0 +1,794 @@
+
+	return res, nil
+}
+
+func (f *Fs) newObject(ctx context.Context, remote string, file client.File) *Object {
+	remoteFile := strings.TrimLeft(strings.Replace(file.FilePath, f.EncodePath(f.root), "", 1), "/")
+
+	folderPath := f.DecodePath(remoteFile[:strings.LastIndex(remoteFile, "/")+1])
+	fileName := f.DecodeFileName(remoteFile[strings.LastIndex(remoteFile, "/")+1:])
+
+	remoteFile = path.Join(folderPath, fileName)
+
+	if file.Type == "file-version" {
+		remoteFile = version.Add(remoteFile, file.UpdatedAt)
+
+		return &Object{
+			fs:          f,
+			remote:      remoteFile,
+			filePath:    file.FilePath,
+			contentType: file.Mime,
+			timestamp:   file.UpdatedAt,
+			file:        file,
+			versionID:   file.VersionInfo["id"],
+		}
+	}
+
+	return &Object{
+		fs:          f,
+		remote:      remoteFile,
+		filePath:    file.FilePath,
+		contentType: file.Mime,
+		timestamp:   file.UpdatedAt,
+		file:        file,
+	}
+}
+
+// NewObject finds the Object at remote.  If it can't be found
+// it returns the error ErrorObjectNotFound.
+//
+// If remote points to a directory then it should return
+// ErrorIsDir if possible without doing any extra work,
+// otherwise ErrorObjectNotFound.
+func (f *Fs) NewObject(ctx context.Context, remote string) (fs.Object, error) {
+	r := path.Join(f.root, remote)
+
+	fs.Debugf(r, "Getting object")
+
+	folderPath := f.EncodePath(r[:strings.LastIndex(r, "/")+1])
+	fileName := f.EncodeFileName(r[strings.LastIndex(r, "/")+1:])
+
+	isFolder := f.getFolderByName(ctx, folderPath, fileName)
+
+	if isFolder != nil {
+		return nil, fs.ErrorIsDir
+	}
+
+	file := f.getFileByName(ctx, folderPath, fileName)
+
+	if file == nil {
+		return nil, fs.ErrorObjectNotFound
+	}
+
+	return f.newObject(ctx, r, *file), nil
+}
+
+// Put in to the remote path with the modTime given of the given size
+//
+// When called from outside an Fs by rclone, src.Size() will always be >= 0.
+// But for unknown-sized objects (indicated by src.Size() == -1), Put should either
+// return an error or upload it properly (rather than e.g. calling panic).
+//
+// May create the object even if it returns an error - if so
+// will return the object and the error, otherwise will return
+// nil and the error
+func (f *Fs) Put(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (fs.Object, error) {
+	return uploadFile(ctx, f, in, src.Remote(), options...)
+}
+
+// Mkdir makes the directory (container, bucket)
+//
+// Shouldn't return an error if it already exists
+func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+	remote := path.Join(f.root, dir)
+
+	fs.Debugf(remote, "Creating directory")
+
+	parentFolderPath := f.EncodePath(remote[:strings.LastIndex(remote, "/")+1])
+	folderName := f.EncodePath(remote[strings.LastIndex(remote, "/")+1:])
+
+	_, e := f.ik.CreateFolder(ctx, client.CreateFolderParam{
+		ParentFolderPath: parentFolderPath,
+		FolderName:       folderName,
+	})
+
+	if e != nil {
+		return e
+	}
+
+	return nil
```

Comment:
Just `return e`? But please call it `err` for consistency.

There are more below which I won't note.